### DISCUSSION TITLE
Why eslint, react types and typescript is in the dependencies and not dev dependencies?

### DISCUSSION THREAD

**Discussion by thevipinmishra** [vercel/next.js#43277] [Help] [answered]
With the changes in the `create next-app`, now all the dependencies which used to be in the devDependencies is now in the dependencies directly.

Why is this intentional ?

```js
  "dependencies": {
    "@types/node": "18.11.9",
    "@types/react": "18.0.25",
    "@types/react-dom": "18.0.9",
    "eslint": "8.28.0",
    "eslint-config-next": "13.0.4",
    "next": "13.0.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "typescript": "4.9.3"
  },
```

**Comment by icyJoseph** [comment, marked as answer]
I think it is because they really want to start pushing forward the output standalone mode:

https://github.com/vercel/next.js/pull/42218

Though I wonder if they considered `npm run audit` situations, where we add `--production` to the command, to only run it on `dependencies` and avoid having our CI/CD complain about a 4th level dependency of `some-random-dev-Dep` having some CVE, which we know doesn't impact our codebase.

That being said, when it comes to this type of apps, rather than packages or libraries, there is a blurry division when it comes to dependencies and devDependencies which I am not going to jump into, in the interest of time (it might explode into an actual lengthy article). And you could just rearrange the `package.json` file to use devDependencies :)

**Comment by thevipinmishra** [comment]
I guess it was all fine if one had to do `npm install` manually.

or they could simply add a prompt to confirm `npm install`ing and we no longer need to do this dependencies merge.

It doesn't really matter at the end of the day, and as you suggested, one can also re-arrange the `package.json` but having everything in dependencies felt "weird".

**Reply by hyochan** [reply]
Agree with @thevipinmishra. I think the abstraction being done with `output standalone mode` may scare the developers into unwanting behavior. For example, we do not want to feel afraid when using `nodemon` and being deployed or not.
The anxiety may be temporary, but for now, you may spend more time trying to understand the abstraction to completely shake off this anxiety.
